test/object-store/realm.cpp

```diff
@@ -358,6 +358,38 @@ TEST_CASE("SharedRealm: get_shared_realm()") {
         REQUIRE(old_realm->schema().size() == 1);
     }
 
+    SECTION("should read the proper schema from the file if realm is frozen") {
```

Comment:
This test should fail if you comment out these lines: https://github.com/realm/realm-core/pull/5784/files#diff-55d86000317cc91bb15c4dfb558cd47af4e8dadecbfc7f5eef4524532a054fcbR257-R260
As it stands, that case still isn't covered.